pkg/interfaces/contracts/solidity-utils/misc/ICowConditionalOrder.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.8.0 <0.9.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+struct GPv2Order {
+    IERC20 sellToken;
+    IERC20 buyToken;
+    address receiver;
+    uint256 sellAmount;
+    uint256 buyAmount;
+    uint32 validTo;
+    bytes32 appData;
+    uint256 feeAmount;
+    bytes32 kind;
+    bool partiallyFillable;
+    bytes32 sellTokenBalance;
+    bytes32 buyTokenBalance;
+}
+
+/**
+ * @title Conditional Order Interface
+ * @author CoW Protocol Developers + mfw78 <mfw78@rndlabs.xyz>
+ */
+interface ICowConditionalOrder {
+    /// @dev This error is returned by the `getTradeableOrder` function if the order condition is not met.
+    ///      A parameter of `string` type is included to allow the caller to specify the reason for the failure.
+    error OrderNotValid(string);
+
+    // --- errors specific for polling
+    // Signal to a watch tower that polling should be attempted again.
+    error PollTryNextBlock(string reason);
+    // Signal to a watch tower that polling should be attempted again at a specific block number.
+    error PollTryAtBlock(uint256 blockNumber, string reason);
+    // Signal to a watch tower that polling should be attempted again at a specific epoch (unix timestamp).
+    error PollTryAtEpoch(uint256 timestamp, string reason);
+    // Signal to a watch tower that the conditional order should not be polled again (delete).
+    error PollNever(string reason);
+
+    /**
+     * @dev This struct is used to uniquely identify a conditional order for an owner.
+     *      H(handler || salt || staticInput) **MUST** be unique for an owner.
```

Comment:
```suggestion
     * H(handler || salt || staticInput) **MUST** be unique for an owner.
```
Dev text isn't justified. Also, type declarations go on the very top, then state variables, events, errors, modifiers, and functions.